### DISCUSSION TITLE
Buy bot

### DISCUSSION THREAD

**Discussion by mertf** [coccoinomane/crabada.py#41] [Ideas]
What do you guys think about implementing a buyer bot from the marketplace? The bot would automatically buy listed crabs whose criteria we setup and under certain amount we determine in the config file. 

For example; 
Bulk
Purity = 6
Breed <= 4
Price <= 14500

**Comment by coccoinomane** [comment]
It would not be hard to add marketplace capabilities to crabada.py.

You could start by creating a `MarketplaceWeb3Client` class which implements the endpoints of the marketplace.

I am planning to add the ABI json of the marketplace in the next few days, so stay tuned.

Then you need to make your listening function in the `bot/marketplace` folder.
Your script can be placed in the `bin/marketplace` folder.

I had thought about doing it in the past, but alas I lack the funds to make it really profitable 😅
